pkg/router/router_test.go

```diff
@@ -277,6 +440,12 @@ func (e mustCreate) Apply(h *harness) error {
 	return err
 }
 
+type mustCreateWithConfig struct {
+	mustCreate
+	configPattern string
```

Comment:
The name is confusing; the test actually performs a substring search, not a pattern match (such as a regexp), right?

Reply:
Didn't have many idea for the name, renamed to `configSnippet`. Tell me if you have better ideas.